tests/test_auth.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from unittest.mock import patch
+from sdk import SDK
+
+class TestAuth:
+    @patch("sdk.client.getpass")
+    def test_module_getpass_success(self, getpass):
+        getpass.return_value = "password"
+        SDK(group_id = "random", username="hi")
+        getpass.assert_called_once()
+
+    @patch("sdk.client.getpass")
+    def test_module_getpass_no_password(self, getpass):
+        getpass.return_value = ""
+        
+        with pytest.raises(Exception):
```

Comment:
Can we catch a more specific exception here ?

Reply:
I can specify a more adequate error indeed